src/components/dashboard/dashboard.service.ts

```diff
@@ -576,27 +576,76 @@ export class DashboardService implements OnDestroy {
     }
 
     /**
-     * When dragging any widgets that need to be moved should be moved to an appropriate position
+     * Return the set of widgets which overlap the given dashboard region.
      */
-    shiftWidgets(): void {
-
+    getOverlappingWidgets(region: DashboardRegion, actionWidget: DashboardWidgetComponent): DashboardWidgetComponent[] {
         let widgetsToMove: DashboardWidgetComponent[] = [];
 
-        const placeholder = this.placeholder$.getValue();
-
-        // check if there are any widgets under the placeholder
-        for (let row = placeholder.row; row < placeholder.row + placeholder.rowSpan; row++) {
-            for (let column = placeholder.column; column < placeholder.column + placeholder.columnSpan; column++) {
+        // check if there are any widgets overlapping widgets
+        for (let row = region.row; row < region.row + region.rowSpan; row++) {
+            for (let column = region.column; column < region.column + region.columnSpan; column++) {
 
                 // store reference to any widgets that need moved
                 this.getOccupiedSpaces()
-                    .filter(space => space.column === column && space.row === row && space.widget !== this._actionWidget.widget)
+                    .filter(space => space.column === column && space.row === row && space.widget !== actionWidget)
                     .forEach(space => widgetsToMove.push(space.widget));
             }
         }
 
         // remove any duplicates
-        widgetsToMove = widgetsToMove.filter((widget, idx, array) => array.indexOf(widget) === idx);
+        return widgetsToMove.filter((wgt, idx, array) => array.indexOf(wgt) === idx);
+    }
+
+
+    /**
+     * Resolve any overlapping widgets after a widget changes rowSpan/colSpan.
+     */
+    resizeWidget(widget: DashboardWidgetComponent): void {
+
+        try {
+
+            // make widget action and origin widget, direction is irrelevant to this function so set to 0
+            this._actionWidget = { widget: widget, direction: 0 };
+            this._widgetOrigin = widget;
+
+            const widgetRegion = {
+                row: widget.row,
+                column: widget.col,
+                rowSpan: widget.rowSpan,
+                columnSpan: widget.colSpan
+            };
+
+            let i = 0;
+            let limit = 100;
```

Comment:
I would probably move this outside of the function (probably outside of the component as well) so that it doesn't go on the stack each time this is called. I tend to use all caps for actual constants.
```ts
const ITERATION_LIMIT = 100;
```